code/modules/reagents/chemistry/holder.dm

```diff
@@ -61,163 +61,174 @@ var/const/INJECT = 5 //injection
 /datum/reagents/Destroy()
 	. = ..()
 	STOP_PROCESSING(SSobj, src)
-	var/list/cached_reagents = reagent_list
-	for(var/reagent in cached_reagents)
-		var/datum/reagent/R = reagent
-		qdel(R)
-	cached_reagents.Cut()
-	cached_reagents = null
+	if(reagent_list)
+		var/list/cached_reagents = reagent_list
+		for(var/reagent in cached_reagents)
+			var/datum/reagent/R = reagent
+			qdel(R)
+		cached_reagents.Cut()
+		cached_reagents = null
 	if(my_atom && my_atom.reagents == src)
 		my_atom.reagents = null
 
 /datum/reagents/proc/remove_any(amount = 1)
-	var/list/cached_reagents = reagent_list
-	var/total_transfered = 0
-	var/current_list_element = 1
+	if(reagent_list)
```

Comment:
don't place an entire proc inside of an if. do `if(!reagent_list) return`

https://github.com/tgstation/tgstation/blob/master/CONTRIBUTING.md#use-early-return